server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -637,27 +631,79 @@ private List<String> createClusterChain(final Map<String, ClusterMetadataManifes
             return Collections.emptyList();
         }
         if (validClusterUUIDs.size() > 1) {
-            throw new IllegalStateException(
-                String.format(
-                    Locale.ROOT,
-                    "The system has ended into multiple valid cluster states in the remote store. "
-                        + "Please check their latest manifest to decide which one you want to keep. Valid Cluster UUIDs: - %s",
-                    validClusterUUIDs
-                )
+            final Map<String, ClusterMetadataManifest> manifestsByClusterUUIDTrimmed = trimClusterUUIDs(
+                manifestsByClusterUUID,
+                validClusterUUIDs,
+                clusterName
             );
+            if (manifestsByClusterUUID.size() == manifestsByClusterUUIDTrimmed.size()) {
```

Comment:
Here this check means that we were not able to trim any clusterUUID. So if we create the chain again, it will again create multiple chains which is invalid.
After trim we call the same function recursively. If a single valid chain was created then this block will not be executed.